scripts/install_dev_tools.sh

```diff
@@ -0,0 +1,211 @@
+
+    # Little hack for grep as some projects are shipping release tarballs
+    # as x86_64 and some as amd64
+    if [[ "${arch}" == "x86_64" ]]; then
+        arch="-e ${arch} -e amd64"
+    elif [[ "${arch}" == "amd64" ]]; then
+        arch="-e ${arch} -e x86_64"
+    fi
+
+    # Call the API to find the latest binary matching kernel and architecture
+    local url_cmd
+    local download_url
+    url_cmd="curl -s ${api_url} | \
+          grep -o -E 'https://(.*).tar.gz' | \
+          grep -i '${kernel_name}' | \
+          grep ${arch}"
+    download_url=$(eval "${url_cmd}")
+
+    if [ -z "${download_url}" ]; then
+        echo "get_download_url_from_github(): download_url not available." >&2
+        echo "command used to get URL: $ ${url_cmd}" >&2
+        return 2
+    fi
+
+    # Return URL
+    echo "${download_url}"
+}
+
+# Installs helm into .venv path
+function install_helm() {
+    local helm_install_script=$1
+    local dest_dir=$2
+    if [ ! -f "${helm_install_script}" ]; then
+        echo "install_helm(): File does not exists: ${helm_install_script}" >&2
+        return 2
+    fi
+
+    if [ ! -d "${dest_dir}" ]; then
+        echo "install_helm(): Dest dir does not exists: ${dest_dir}" >&2
+        return 2
+    fi
+
+    chmod +x "${helm_install_script}" 
+    HELM_INSTALL_DIR="${dest_dir}" USE_SUDO=false "${helm_install_script}"
+}
+
+# Function to safely remove temporary files and temporary download dir
+# Argument is optional exit value to propagate it after cleanup
+function cleanup_and_exit() {
+    local exit_val=$1
+    if [ -z "${DWN_DIR}" ]; then
+        echo "cleanup_and_exit(): Temp download dir not provided !" >&2
+    else
+      # Ensure dir exists and starts with prefix
+      if [ -d "${DWN_DIR}" ]; then
+          PELORUS_TMP_DIR=$(basename "${DWN_DIR}")
+          if [[ "${PELORUS_TMP_DIR}" =~ "${TMP_DIR_PREFIX}"* ]]; then
+              echo "Cleaning up temporary files"
+              eval rm -f "${DWN_DIR}/*"
+              rmdir "${DWN_DIR}"
+          fi
+      fi
+    fi
+    # Propagate exit value if was provided
+    [ -n "${exit_val}" ] && exit "$exit_val"
+    exit 0
+}
+
+function print_help() {
+    printf "\nUsage: %s [OPTION]... -v [DIR]\n\n" % "$0"
+    printf "\tStartup:\n"
+    printf "\t  -h\tprint this help\n"
+    printf "\n\tOptions:\n"
+    printf "\t  -v\tpath to virtualenv DIR\n"
+
+    exit 2
+}
+
+### Options
+OPTIND=1
+while getopts "h?v:" option; do
+    case "$option" in
+    h|\?) print_help;;
+    v)    venv_dir=$OPTARG;;
+    esac
+done
+
+if [ -z "${venv_dir}" ]; then
+    VENV="${DEFAULT_VENV}"
+else
+    VENV="${venv_dir}"
+fi
+
+# Create download directory inside virtual env dir
+DWN_DIR=$(mktemp -p "${VENV}" -d -t "${TMP_DIR_PREFIX}XXXXX") || exit 2
+
+# Helm install
+if ! [ -x "$(command -v "${VENV}/bin/helm")" ]; then
+    echo "Installing helm CLI to: ${VENV}/bin/helm"
+    download_file_from_url "${HELM_INSTALL_SCRIPT}" "${DWN_DIR}" || cleanup_and_exit 2
```

Comment:
Instead of appending every line with the cleanup_and_exit call, we can set a [signal trap](https://mywiki.wooledge.org/SignalTrap) to do the cleanup. This'll do the cleanup even if there's an exit condition we didn't catch, and even handle the happy path.

Reply:
definatelly !